### DISCUSSION TITLE
sha256 does not provide an export named 'default'

### DISCUSSION THREAD

**Discussion by lissergey** [supabase/supabase#13340] [Questions] [answered]
Hello. Deploying a new nuxt 3 and supabase project. I'm trying to implement user authorization and I get an error when loading the page. 
I tried deploying a new project on another device, and so on, nothing helps.

`Uncaught SyntaxError: The requested module '/_nuxt/node_modules/crypto-js/sha256.js?v=a4564608' does not provide an export named 'default'`

Error linked to _nuxt/node_modules/@supabase/gotrue-js/src/lib/helpers.ts and line with this code
`import sha256CryptoJS from 'crypto-js/sha256'`

**Comment by rscorer** [comment]
I just got the same.

Nuxt : 3.3.2
Supabase: 0.3.1

**Comment by jaycelcunanan** [comment]
Also having same issue

Cannot find module '/opt/render/project/src/.output/server/node_modules/crypto-js/sha256'

Nuxt : 3.3.2
Supabase: 0.3.1

**Comment by alexryans** [comment]
Same issue here - `Uncaught SyntaxError: The requested module '/_nuxt/node_modules/crypto-js/sha256.js?v=eecd9645' does not provide an export named 'default' (at helpers.ts:2:8)`

Nuxt: 3.3.2
Supabase: 0.3.1

**Comment by elonsalfati** [comment]
```json
{
  "@nuxtjs/supabase": "^0.3.1",
  "nuxt": "^3.3.2",
}
```

Yep - same here. I've been banging my head on the wall for the past 24 hours.

I feel like this should be an open bug - @thorwebdev @awalias

**Reply by elonsalfati** [reply]
I have opened [this PR](https://github.com/supabase/gotrue-js/pull/640) that solves the issue.

@lissergey @rscorer @jaycelcunanan @alexryans

**Reply by yldshv** [reply]
> I have opened [this PR](https://github.com/supabase/gotrue-js/pull/640) that solves the issue.
> 
> @lissergey @rscorer @jaycelcunanan @alexryans

Sorry im a beginner is there a way to fix this bug locally in project i dont want to wait until this is officialy fixed.
Maybe there is a older version? I tried with 0.3.0 but i got the same error.
Ty

**Reply by elonsalfati** [reply]
Hi @yldshv 

You can use my PR branch in the meantime by overriding the default `@supabase/gotrue-js` dependency.

![image](https://user-images.githubusercontent.com/22857167/229478796-02a665f4-cc7d-42f3-87dc-31471bf8aeec.png)

Please keep in mind that I'm not planning on rebasing or keeping this branch up to date. So, once my PR is merged, or supabase introduces a different solution - move to their option.

**Comment by J0** [comment]
Hey all,

Thanks for catching this bug and supplying a fix - this is an issue resulting from a dependency (crypto-js) introduced between v2.8.0 and v2.13.0  in order to support the PKCE flow. We're looking into the PR mentioned above and are testing it now on a  few environments to ensure that the fix is compatible across the environments we support. 

In the meantime, do run npm install '@supabase/supabase-js@2.8.0' to fetch a version of @supabase/supabase-js which doesn't have `crypto-js` included.

Once again, apologies for the inconvenience caused and appreciate your patience with us.

Thanks!
Jo

**Reply by samuel-antwi** [reply]
Has anyone managed to get this working? I can confirm fetching @supabase/supabase-js@2.8.0 still doesn't solve the issue.

**Reply by estige** [reply]
Here not work, adding version 2.8.0 of supabase-js.

**Reply by silentworks** [reply]
You would need to delete your `node_modules` directory before doing the downgrade. We will release a new version in the next few days that should resolve this all together in new installs.

**Reply by silentworks** [reply]
This should be updated in the latest version now.

**Comment by thorwebdev** [comment, marked as answer]
This has been fixed, sorry for the inconvenience caused: https://github.com/supabase/gotrue-js/releases/tag/v2.18.1
